src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py

```diff
@@ -192,6 +192,27 @@ def test_download_linux_web_log(self, resource_group):
         self.assertTrue(os.path.isdir(os.path.join(log_dir, 'LogFiles', 'kudu', 'trace')))
 
 
+class WebappZipDeployScenarioTest(ResourceGroupVCRTestBase):
+    def __init__(self, test_method):
+        super(WebappZipDeployScenarioTest, self).__init__(__file__, test_method, resource_group='cli-webapp-zipDeploy')
+        self.webapp_name = 'azurecli-webapp-zipdeploy'
+
+    def test_deploy_zip(self):
+        self.execute()
+
+    def body(self):
+        plan = 'webapp-zip-deploy-plan'
+        zip_file = os.path.join(TEST_DIR, 'test.zip')
```

Comment:
Please do not check in a binary file to the git repository. You can check in the source code of the sample project and create a zip file from it on-the-fly.

Reply:
IMO it's okay for a test.
The zip file is 200Bytes and has a single html file in and the point is that the test is for uploading a zip file.
Zipping up the HTML for each run of the test seems odd and if a different OS archives the zip, differently, the test would behave differently.

Reply:
Can i know what the concerns are regarding adding a .zip file? all this has is a simple html file. creating a zip file on the fly is not an accurate test here.

Reply:
Generally speaking, it is discouraged to add binary files to the git repo since each copy of the binary file will be carried forever in the repository. Therefore the large binary file will quickly lead to swollen git repositories. Though @derekbekoe made a good point here that this file is not that big, we do like to avoid slip slop effect (as @panchagnula correctly pointed out in the `sleep` method comment that there were bad practices in the repository which cause broken window effect.)

Python standard library offeres [zipfile](https://docs.python.org/2.7/library/zipfile.html?highlight=zip) model to generate compressed package. It is cross-platform. It is a consistent approach to generate the zip.